plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1475,6 +1494,19 @@ public void finishStatisticsCollection(ConnectorSession session, ConnectorTableH
         }
     }
 
+    private static List<String> canonicalizePartitionValues(String partitionName, List<String> partitionValues, List<Type> partitionTypes)
+    {
+        verify(partitionValues.size() == partitionTypes.size(), "Expected partitionTypes size to be %s but got %s", partitionValues.size(), partitionTypes.size());
+        ImmutableList.Builder<Block> parsedPartitionValuesBuilder = ImmutableList.builder();
```

Comment:
use plain `Block[]`

Reply:
can we avoid Block overhead?

Reply:
> can we avoid Block overhead?

I tend to say no. See https://github.com/trinodb/trino/blob/19f61fc56984bd98c617d8e72214a61237f8b256/plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveWriteUtils.java#L301

I was looking also at 

https://github.com/trinodb/trino/blob/19f61fc56984bd98c617d8e72214a61237f8b256/core/trino-spi/src/main/java/io/trino/spi/type/Type.java

and haven't found a way to retrieve the object value without a `Block`.

Reply:
https://github.com/trinodb/trino/blob/0954991016d98487d1ff718ad7aa86ac5d48edb8/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/MetastoreUtil.java#L389 

looks promissing, but doesn't quite have the complexity of 

https://github.com/trinodb/trino/blob/0954991016d98487d1ff718ad7aa86ac5d48edb8/plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveUtil.java#L556


I don't find it a good fit.

Reply:
Does it make sense to avoid canonicalization when the computed statistics contain the key?

Reply:
> Does it make sense to avoid canonicalization when the computed statistics contain the key?

Are you implying that we could canonicalize the partition values only if the list of "raw" partition values is not found within the computed statistics map?

Reply:
Yes, that's what I suggested. e.g. https://github.com/trinodb/trino/pull/15995/files#diff-1e06d5ca5f4d2b8a7964d589fbe0278d6aa9d99086c4e5ce2c4fb974b9184b20R1482 
```java
                ComputedStatistics collectedStatistics = computedStatisticsMap.containsKey(partitionValues)
                        ? computedStatisticsMap.get(partitionValues)
                        : computedStatisticsMap.get(canonicalizePartitionValues(partitionName, partitionValues, partitionTypes));
```